src/string_id.h

```diff
@@ -208,14 +273,31 @@ class string_id
             return !is_null();
         }
 
+        uint64_t hash() const {
+            return _hash;
+        }
+
     private:
-        std::string _id;
+        std::map < uint64_t, std::unique_ptr< std::string > > &cached_strings() const {
+            static std::map < uint64_t, std::unique_ptr< std::string > > cached_strings_{};
```

Comment:
"static" `cached_strings_` here will actually be a separate collection for every `string_id.h` header inclusion X every id type.  Was that your intention?

Reply:
Yes, to make the emplaces faster on each of them, as the large majority of strings won't appear in more than one cached_strings object (Barring the obvious null and the such)